src/simulations/collect.jl

```diff
@@ -115,14 +115,25 @@ add_data(s, when::Bool) = when
 add_data(s, when) = when(s)
 
 function _step!(
-    model, agent_step!, model_step!, n, when = true;
-    model_properties=nothing, aggregation_dict=nothing, agent_properties=nothing
+    model, agent_step!, model_step!, n;
+    collect0 = true, when = true,
+    model_properties=nothing, aggregation_dict=nothing, agent_properties=nothing,
   )
-  df_agent = DataFrame()
-  df_model = DataFrame()
+  if collect0
```

Comment:
I find `collect0` unnecessary. The user can just choose to include 0 in when or not.

Reply:
That is a good point, but it doesn't work if `when` is a function instead of numbers.